pykoplenti/__init__.py

```diff
@@ -401,22 +402,32 @@ def _create_url(self, path: str) -> URL:
         )
         return base.join(URL(path))
 
-    async def login(self, password: str, user: str = "user"):
-        """Login the given user (default is 'user') with the given
-        password.
+    async def login(self,
+                    key: str,
+                    service_code: Union[str, None] = None):
```

Comment:
I understand the decision to rename and remove of the login arguments, but for the sake to stay compatible with the current version I would suggest the following:
* Use `password` instead of `key`.
* The user parameter should still be accepted (either directly or via `*args`), but set to `None` as default.

Reply:
I understand your request. In general I would have kept the function signature backward-compatible, but in this particular case a breaking change is justifiable. 

Allowing to configure a user name is misleading and may result in an inconsistent parameterization of the `login` method (user: 'master', password: 'abc', service_code: None). From all what we know about the API of the inverter (including the reverse engineered web interface) it only supports two users names: "user" and "master".  

The parameter name `key` (as in Master Key) seemed closer to the authentication concept to me than `password`, but I would be willing to revert this name change to minimize the impact of my changes (eventhough the parameter is positional). 

Since this isn't a bugfix, but a new feature, I'd like to propose to keep my changes (with or without renaming `key` to `password`) on a dev branch for a future major version release. Would this be an acceptable compromise?

Reply:
What about the following, backwards compatible solution:

```suggestion
    async def login(self, 
                    key: str, 
                    service_code: Union[str, None] = None,
                    password: Union[str, None] = None, 
                    user: Union[str, None] = None):
        if password is None:
            self._key = key
        else:
            warnings.warn("password is deprecated. Use key instead.", DeprecationWarning)
            self._key = password

        self._key = key if password is None else password
        if user is None:
            self._user = 'master' if service_code else 'user'
        else:
            warnings.warn("user is deprecated. user is chosen automatically.", DeprecationWarning)
            self._user = user
```
Works if
* password is specified positional (it uses than the key argument)
* password is specified as keyword argument - warning is printed
* user is specified - warning is printed
* the new arguments are used

Reply:
In your proposal the parameter `key` should get a default value and line 416 duplicates the if statement above.

I guess a cleaner alternative would be to introduce a new login method, for example named `authenticate`, and keep the old `login` method as it is, adding a deprecation notice to the docstring - not as runtime warning.

In the end you are the maintainer, so please feel free to pick the solution you expect the users to value most. 

Many thanks!